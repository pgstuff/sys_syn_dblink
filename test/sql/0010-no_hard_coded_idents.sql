BEGIN;

CREATE EXTENSION sys_syn_dblink;

CREATE SCHEMA processor_schema
        AUTHORIZATION postgres;

SELECT  dblink_connect('sys_syn_data', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)));
SELECT dblink_exec('sys_syn_data', 'BEGIN');

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('sys_syn_data', 'group');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('sys_syn_data', 'group');
INSERT INTO sys_syn_dblink.put_groups_def VALUES ('put_group');

SELECT sys_syn_dblink.processing_table_create (
        proc_schema     => 'processor_schema',
        in_table_id     => 'test_data',
        out_group_id    => 'group',
        put_group_id    => 'put_group',
        dblink_connname => 'sys_syn_data');


SELECT * FROM processor_schema.test_data_group_claim();

SELECT * FROM processor_schema.test_data_group_pull();

SELECT * FROM processor_schema.test_data_group_process();

SELECT * FROM processor_schema.test_data_group_push_status();


SELECT dblink_exec('sys_syn_data', 'ROLLBACK');
SELECT dblink_disconnect('sys_syn_data');

ROLLBACK;
