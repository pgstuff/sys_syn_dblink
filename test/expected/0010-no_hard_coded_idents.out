BEGIN;
CREATE EXTENSION sys_syn_dblink;
CREATE SCHEMA processor_schema
        AUTHORIZATION postgres;
SELECT  dblink_connect('sys_syn_data', 'dbname=contrib_regression host=' ||
        quote_literal(split_part((SELECT pg_settings.setting FROM pg_settings WHERE pg_settings.name = 'unix_socket_directories'), ', ', 1)));
 dblink_connect 
----------------
 OK
(1 row)

SELECT dblink_exec('sys_syn_data', 'BEGIN');
 dblink_exec 
-------------
 BEGIN
(1 row)

INSERT INTO sys_syn_dblink.in_groups_def VALUES ('sys_syn_data', 'group');
INSERT INTO sys_syn_dblink.out_groups_def VALUES ('sys_syn_data', 'group');
INSERT INTO sys_syn_dblink.put_groups_def VALUES ('put_group');
SELECT sys_syn_dblink.processing_table_create (
        proc_schema     => 'processor_schema',
        in_table_id     => 'test_data',
        out_group_id    => 'group',
        put_group_id    => 'put_group',
        dblink_connname => 'sys_syn_data');
 processing_table_create 
-------------------------
 
(1 row)

SELECT * FROM processor_schema.test_data_group_0_claim();
 test_data_group_0_claim 
-------------------------
 t
(1 row)

SELECT * FROM processor_schema.test_data_group_0_pull();
 test_data_group_0_pull 
------------------------
 t
(1 row)

SELECT * FROM processor_schema.test_data_group_0_process();
 test_data_group_0_process 
---------------------------
 t
(1 row)

SELECT * FROM processor_schema.test_data_group_0_push_status();
 test_data_group_0_push_status 
-------------------------------
 t
(1 row)

SELECT dblink_exec('sys_syn_data', 'ROLLBACK');
 dblink_exec 
-------------
 ROLLBACK
(1 row)

SELECT dblink_disconnect('sys_syn_data');
 dblink_disconnect 
-------------------
 OK
(1 row)

ROLLBACK;
